dashboards-observability/public/components/explorer/home_table/history_table.tsx

```diff
@@ -40,43 +39,100 @@ export function Table(options: TableData) {
   };
 
   const columns = [
+    {
+      field: 'type',
+      name: '',
+      width: '3%',
+      render: (item) => {
+        if (item == 'Visualization') {
+          return (
+            <div>
+              <EuiLoadingChart size="m" />
+            </div>
+          );
+        } else {
+          return (
+            <div>
+              <EuiIcon type="discoverApp" size="m" />
+            </div>
+          );
+        }
+      },
+    },
     {
       field: 'data',
       name: 'Name',
-      render: (item)=>{return <EuiLink onClick={() =>
-      {options.savedQuerySearch(item.query, [item.date_start, item.date_end], item.timestamp, item.fields)}}>
-        {item.name}
-      </EuiLink>},
+      align: 'left',
+      render: (item) => {
+        return (
+          <EuiLink
+            onClick={() => {
+              options.savedQuerySearch(
```

Comment:
Maybe a different name? the name savedQuerySearch is a little bit confusing, since it is just a function dispatching data.

Reply:
It's actually dispatching data for the query search, thats why i used the same name. should i still change it?

Reply:
Maybe it's fine, the dispatch will be removed once the changes to support passing object id gets merged.